app.js

```diff
@@ -72,6 +72,7 @@ log('preparing osm indexes..')
 osm.ready(function () {
   log('osm indexes READY')
 })
+app.osm = osm
```

Comment:
At some point I want to re-think how we structure this and the way data flows through the app, borrowing ideas from React, and any other functional UI app. It makes me nervous as we keep threading bits of context between different modules, but it's fine for now.